### PR TITLE
Only log task started once

### PR DESCRIPTION
**Pull request by Forgind**
🤦

Fixes #

### Changes Made
Only log task started once, even when MSBuildEventSource.Log.IsEnabled()

### Testing
None

### CODE REVIEW

**Comment by danmoseley** [comment]
Isnt a test possible!

**Comment by Forgind** [comment]
> Is a test possible?

In theory, yeah, probably, but I think the cost would outweigh the benefit in this case. It's a pretty minor bug anyway, more of an annoyance. The sort of thing that I'm happy fixing but wouldn't bother if I had to do too much extra work beyond just the fix.

**Review by rainersigwald — APPROVED** [review]
Ah! This explains why it was happening "only sometimes": it was happening only when `DOTNET_PERFLOG_DIR` was set, turning on our event source.

I think a test that validates "when this artificial project that has exactly one target and one task is built, the events are exactly and in order A, B, C, D . . ." would be nice. But I won't block this PR on that (because this tripling is so annoying to me 😇).

File: src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs [hunk 1]
```diff
@@ -433,6 +427,10 @@ private async Task<WorkUnitResult> ExecuteBucket(TaskHost taskHost, ItemBucket b
                     if (requirements != null)
                     {
                         TaskLoggingContext taskLoggingContext = _targetLoggingContext.LogTaskBatchStarted(_projectFullPath, _targetChildInstance);
+                        if (MSBuildEventSource.Log.IsEnabled())
```
Comment (ladipro) [inline src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs:430]: nit: The `IsEnabled` check is likely not needed here. Evaluating the arguments is just a few dereferences. I would either remove it or cache the result in a local so it doesn't have to be called twice.

**Review by ladipro — APPROVED** [review]

File: src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs (hunk 1 above)
Comment (Forgind) [reply src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs:430]: Agreed; this was a more complex call before when I first added it.

That said, I don't think it matters. My understanding is that the check is so fast that should basically consider it a no-op as far as time taken, and ExecuteTaskStart implicitly calls IsEnabled after it's evaluated its arguments, so they should be equivalent from a correctness perspective. I'll still remove it. Thanks!

**Comment by Forgind** [comment]
Just noting the first failing leg seems to be due to the CPV change.

**Comment by rainersigwald** [comment]
> Just noting the first failing leg seems to be due to the CPV change.

I don't agree, I think it was an intermittent Azure DevOps artifacts failure.

/azp run

**Comment by Forgind** [comment]
I unfortunately can't see the error message anymore, but if I remember correctly, it was something that seemed like an intermittent AzDO artifacts failure that I'd never seen before the CPV change but appeared multiple times since then...I don't think the change itself was wrong, but I'm wondering if the infrastructure supporting it is flaky.

**Pull request merged**